Sources/SwiftProtobuf/Decoder.swift

```diff
@@ -43,89 +43,88 @@ import Foundation
 /// do distinguish Enum-valued and Message-valued maps to avoid
 /// polluting the generated Enum and Message types with all of the
 /// necessary generic methods to support this.
-
 public protocol Decoder {
-    // Called by a Oneof when it already has a value and is being
-    // asked to accept a new value.  Some decoders specify that oneof
-    // decoding must fail in this case.  Those decoders throw a
-    // suitable error from this function; other decoders make it a
-    // no-op.  No existing decoder actually needs this to be mutating,
-    // but someone might want to track OneOf conflicts in the future.
-    mutating func handleConflictingOneOf() throws
+  // Called by a Oneof when it already has a value and is being
+  // asked to accept a new value.  Some decoders specify that oneof
```

Comment:
I think "Some decoders..." here should be "Some formats require oneof decoding to fail in this case."   (This behavior is required by the format specification, our "decoder" is just an implementation.)